src/PathProcessor/PathProcessorFiles.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Drupal\file_encrypt\PathProcessor;
+
+use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Defines a path processor to rewrite file URLs.
+ *
+ * As the route system does not allow arbitrary amount of parameters convert
+ * the file path to a query parameter on the request.
+ */
+class PathProcessorFiles implements InboundPathProcessorInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processInbound($path, Request $request) {
+    if (strpos($path, '/encrypt/files/') === 0 && !$request->query->has('file')) {
```

Comment:
OK, let's give it a shot then. I'm just wondering because it seems like "loose" context to me.